cirq/sim/sparse_simulator_test.py

```diff
@@ -66,10 +67,30 @@ def test_run_bit_flips(dtype):
                                     {'0': [[b0]], '1': [[b1]]})
 
 
+class CountingSimulator(cirq.Simulator):
+    """
+    CountingSimulator is a test class that decorates the _base_iterator method
+    so that we can count the number of circuit runs during a test case
+    """
+    def __init__(self, dtype=np.complex64):
+        super().__init__(dtype)
+        self.count = 0
+
+    def _base_iterator(
+            self,
+            circuit: cirq.Circuit,
+            qubit_order: cirq.ops.QubitOrderOrList,
+            initial_state: Union[int, np.ndarray],
+            perform_measurements: bool = True,
+    ):
+        self.count += 1
```

Comment:
Use mocks instead.

(Note: python 2 and python 3 get their mocks from different places, so we have `cirq.testing.mock` to hook up the right one.)

Reply:
ack - thanks for pointing me at cirq.testing.mock.